lib/elixir/test/elixir/kernel/overridable_test.exs

```diff
@@ -176,4 +231,80 @@ defmodule Kernel.OverridableTest do
       """
     end
   end
+
+  test "overrides required callback with behaviour as argument" do
+    assert Kernel.OverridableWithBehaviour.required_callback == "overridden"
+  end
+
+  test "overrides optional callback with behaviour as argument" do
+    assert Kernel.OverridableWithBehaviour.optional_callback == "overridden"
+  end
+
+  test "does not override function that is not a callback for this behaviour" do
+    assert Kernel.OverridableWithBehaviour.not_a_behaviour_callback == "original"
+  end
+
+  test "overrides required macro callback with behaviour as argument" do
+    require Kernel.OverridableWithBehaviour
+
+    assert Kernel.OverridableWithBehaviour.required_macro_callback(true) == "overridden"
+  end
+
+  test "specifies optional macro callback with behaviour as argument" do
+    require Kernel.OverridableWithBehaviour
+
+    assert Kernel.OverridableWithBehaviour.optional_macro_callback(true, true) == "defined optional for the first time"
+  end
+
+  test "undefined module can't be passed as argument to defoverridable" do
+    message = "cannot pass module Bar as argument to defoverridable/1 because it was not defined"
+    assert_raise ArgumentError, message, fn ->
+      Code.eval_string """
+      defmodule Foo do
+        defoverridable Bar
```

Comment:
I understand, this is because different test files might be run in many different processes correct?

I copied the pattern from this existing test: https://github.com/elixir-lang/elixir/blob/master/lib/elixir/test/elixir/kernel/overridable_test.exs#L173

Should I namespace the existing test as well?